go-controller/pkg/ovn/ovn.go

```diff
@@ -740,23 +656,46 @@ func (oc *Controller) WatchPods() {
 					podNs, podName)
 				return
 			}
-			if !oc.ensurePod(oldPod, pod, oc.checkAndSkipRetryPod(pod.UID)) {
+			oc.checkAndSkipRetryPod(pod)
+			if retryEntry := oc.getPodRetryEntry(pod); retryEntry != nil && retryEntry.needsDel != nil {
+				klog.Infof("Detected leftover old pod during new pod add with the same name: %s. "+
+					"Attempting deletion of leftover old...", getPodNamespacedName(pod))
+				if err := oc.removePod(pod); err != nil {
+					oc.recordPodEvent(err, pod)
+					klog.Errorf("Failed to delete pod %s, error: %v",
+						getPodNamespacedName(pod), err)
+					oc.unSkipRetryPod(pod)
+					return
+				}
+				// deletion was a success; remove delete retry entry
+				oc.removeDeleteRetry(pod)
+			}
+			if err := oc.ensurePod(oldPod, pod, oc.checkAndSkipRetryPod(pod)); err != nil {
+				klog.Errorf("Failed to update pod %s, error: %v",
```

Comment:
need to record event here too right? oc.recordPodEvent(err, pod)

Reply:
yes, done. specially since I removed it from ensurePod.